### ISSUE TITLE
Introduce configuration option for force shutdown client/server

### ISSUE THREAD

**Issue by asarkar** [yidongnan/grpc-spring-boot-starter#396]
In a regular application without using this library, we would do something like this in the client code to shutdown the `ManagedChannel`:
```
channel
  .shutdown()
  .awaitTermination(5, TimeUnit.SECONDS)
```

 and on the serve side:
```
server.shutdown()
  .awaitTermination(5, TimeUnit.SECONDS)
```

What is the recommended way to release resources when using this library?

**Comment by ST-DDT** [comment]
This library takes care of it:

https://github.com/yidongnan/grpc-spring-boot-starter/blob/master/grpc-client-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/client/channelfactory/AbstractChannelFactory.java#L300

https://github.com/yidongnan/grpc-spring-boot-starter/blob/master/grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/serverfactory/GrpcServerLifecycle.java#L111

**Comment by asarkar** [comment]
> If the channels don't have terminated after 60 seconds then they will be forcefully shutdown

In my integration tests, I'm noticing a long time for them to finish _after_ my test code is executed. Could this be because of the above?

It seems the wait time is neither configurable, nor avoidable. In tests, I might not want to wait as long as 60 seconds. What do you think about introducing two new parameters for the client, `shutdownWaitTime` and `shutdownNow`? In the second case, `shutdownWaitTime` would be zero, no matter what the user set it to.

**Comment by ST-DDT** [comment]
> > If the channels don't have terminated after 60 seconds then they will be forcefully shutdown
> 
> In my integration tests, I'm noticing a long time for them to finish _after_ my test code is executed. Could this be because of the above?

Do you see a lot of `Awaiting channel shutdown:` message in your log?

If yes, then there is still a call running on that channel.
If not, then you might not cause spring/the channel factory to shutdown. Try `@DirtiesContext`

> It seems the wait time is neither configurable, nor avoidable. In tests, I might not want to wait as long as 60 seconds. What do you think about introducing two new parameters for the client, shutdownWaitTime and shutdownNow? In the second case, shutdownWaitTime would be zero, no matter what the user sets it to.

I agree, that such a config option might be desirable. However, I think a single property should be enough. Because if it is set to `0` it will shutdown now.

> The same settings could be applied to the server for consistency.

I agree.

**Comment by asarkar** [comment]
> if it is set to 0 it will shutdown now
Makes sense.

> I'm not sure whether I will add it to the current or the next milestone

That's ok, thanks for accepting the change.

**Comment by asarkar** [comment]
> Try @DirtiesContext

I had, set for after every test method. Removing it helped tremendously.

**Comment by ST-DDT** [comment]
Will be implemented in #491
